libcudacxx/include/cuda/std/__atomic/types/reference.h

```diff
@@ -0,0 +1,61 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _LIBCUDACXX___ATOMIC_TYPES_REFERENCE_H
+#define _LIBCUDACXX___ATOMIC_TYPES_REFERENCE_H
+
+#include <cuda/std/detail/__config>
+
+#include <cuda/std/__atomic/types/base.h>
+#include <cuda/std/type_traits>
+
+_LIBCUDACXX_BEGIN_NAMESPACE_STD
+
+// Reference is compatible with __atomic_base_tag and uses the default dispatch
+template <typename _Tp>
+struct __atomic_ref_storage
+{
+  using __underlying_t                = _Tp;
+  static constexpr __atomic_tag __tag = __atomic_tag::__atomic_base_tag;
+
+#if !defined(_CCCL_COMPILER_GCC) || (__GNUC__ >= 5)
```

Comment:
We do not support GCC 5 anymore, so please drop that conditional

Reply:
We should remove this later.